### PR TITLE
feat: update rules with additional presets

### PR DESCRIPTION
**Pull request by bnb**
adds some additional presets for some rules that are nice-to-haves. has an accompanying PR to the style guide here: https://github.com/nodejs/node/pull/39551

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -46,13 +46,18 @@ module.exports.plugins = [
   [require("remark-lint-first-heading-level"), 1],
   [require("remark-lint-heading-style"), "atx"],
   [require("remark-lint-list-item-indent"), "space"],
+  [require('remark-lint-mdash-style'), '-'],
```
Comment (nschonni) [inline index.js:49]:
```suggestion
  [require("remark-lint-mdash-style"), "-"],
```

File: index.js [hunk 2]
```diff
@@ -46,13 +46,18 @@ module.exports.plugins = [
   [require("remark-lint-first-heading-level"), 1],
   [require("remark-lint-heading-style"), "atx"],
   [require("remark-lint-list-item-indent"), "space"],
+  [require("remark-lint-mdash-style"), "-"],
```
Comment (Trott) [inline index.js:49]: This is currently saying "use an en dash for your em dashes" and I'm pretty opposed to that.

```suggestion
  [require("remark-lint-mdash-style"), "—"],
```

Although I think it's fine for people to use `&mdash;` instead.

File: index.js [hunk 3]
```diff
@@ -46,13 +46,18 @@ module.exports.plugins = [
...
+  require("remark-lint-match-punctuation"),
   require("remark-lint-maximum-line-length"),
   require("remark-lint-no-consecutive-blank-lines"),
+  require("remark-lint-no-dead-urls"),
```
Comment (nschonni) [inline index.js:53]: This one seems to be giving a bunch of false positives

**Comment by Trott** [comment]
I don't think some of these rules are working as advertised. I cannot get the em dash rule to give me an error no matter what I put in the README.md. And the no-dead-urls one provides false positives for me locally as it is doing in the GitHub Action here.

It might be better to add these rules one at a time in separate PRs so that we can pick and choose from the ones that are actually working and that are uncontroversial.

**Comment by Trott** [comment]
@bnb I'm going to close this, but feel free to comment or re-open if you intend to keep working on this.

**Pull request closed without merging**